extensions/spoons/init.lua

```diff
@@ -164,4 +206,152 @@ function module.list(only_loaded)
    return res
 end
 
+-- --------------------------------------------------------------------
+-- Spoon repository management
+
+-- Internal callback to process and store the data from docs.json about a repository
+function _storeRepoJSON(repo, status, body, hdrs)
+   if status < 0 then
+      log.ef("Error fetching JSON data for repository '%s'. Error: %s", repo, body)
+   else
+      local json = hs.json.decode(body)
+      if json then
+         module.repos[repo].data = {}
+         for i,v in ipairs(json) do
+            v.download_url = module.repos[repo].download_base_url .. "/" .. v.name .. ".spoon.zip"
+            module.repos[repo].data[v.name] = v
+         end
+         log.df("Updated JSON data for repository '%s'", repo)
+      else
+         log.ef("Invalid JSON received for repository '%s': %s", repo, body)
+      end
+   end
+end
+
+--- hs.spoons.updateRepo(repo)
+--- Method
+--- Fetch and store locally the information about the contents of a Spoons repository
+---
+--- Parameters:
+---  * repo - name of the repository to update. Defaults to `"default"`.
+---
+--- Returns:
+---  * None
+function module.updateRepo(repo)
+   if not repo then repo = 'default' end
+   if module.repos[repo] and module.repos[repo].url then
+      module.repos[repo].json_url = string.gsub(module.repos[repo].url, "/$", "") .. "/raw/master/docs/docs.json"
+      module.repos[repo].download_base_url = string.gsub(module.repos[repo].url, "/$", "") .. "/raw/master/Spoons/"
```

Comment:
This is where we assume the repo is hosted on GitHub. We could change this if needed, but for now it just makes things simpler. We could in fact shorten the repo URLs to "user/repo" and expand from there. @cmsj What do you think?